### ISSUE TITLE
The aria-labelledby attribute on inner span not being respected

### ISSUE THREAD

**Issue by alexstine** [nvaccess/nvda#13769]
### Steps to reproduce:

I found this bug while browsing around on Facebook. I am not sure if this is related to Facebook or some bug in NVDA, maybe both.
1. Open Facebook and log in. Browser does not matter, can replicate in both Google Chrome and Firefox latest versions.
2. Find a post with the H key for heading.
3. You will land on the posters name.
4. Pressing Down Arrow key will usually read the timestamp but now it reads some random letters and numbers. Instead of "4h", it might read "dtSoornesp8h4ia93h0a8ut16268fm729181fg6haa".
5. Here is a code sample. You can see the link does not have aria-labelledby but the inner span does. I am thinking this is just Facebook and their super smart team of engineers that don't know how to code very well but if aria-labelledby should be respected that deep, wanted to bring it up.

```
<a class="oajrlxb2 g5ia77u1 qu0x051f esr5mh6w e9989ue4 r7d6kgcz rq0escxv nhd2j8a9 nc684nl6 p7hjln8o kvgmc6g5 cxmmr5t8 oygrvhab hcukyx3x jb3vyjys rz4wbd8a qt6c0cv9 a8nywdso i1ao9s8h esuyzwwr f1sip0of lzcic4wl gmql0nx0 gpro0wi8 b1v8xokw" href="https://www.facebook.com/groups/1870565113191997/posts/3128616450720184/?__cft__[0]=AZU8xz0iG78krT0aN_gSujHQxh040YOoDitMY_KKtKsRg35OrdoQ21qt2eGCS9At0aaxy9UUsLcgO_bOeQ3ocaG-CU1byUWK0FxmFAErks3RFurOeX5a2qZIRlfDm30nlP3KjFcECuE_p4zl1nzn7rfZOdlGgZcaX90p7yfgaQGdVsO9psCBT1mqLpYyC4Jwt4s&amp;__tn__=%2CO%2CP-R" role="link" tabindex="0"><span><span class="j1lvzwm4 stjgntxs ni8dbmo4 q9uorilb gpro0wi8" aria-labelledby="jsc_c_48"><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4" style="display: flex; order: 0;"><span class="ihxqhq3m myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 42;">d</span><span class="myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 23;">o</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 12;">p</span><span class="ihxqhq3m myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 2;">t</span><span class="l94mrbxd aenfhxwr myohyog2 b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 40;">e</span><span class="nc684nl6 l94mrbxd l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 9;">S</span><span class="myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 33;">o</span><span class="b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 24;">n</span><span class="nc684nl6 l94mrbxd l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 31;">s</span><span class="t5a262vz l94mrbxd myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 39;">r</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 18;">h</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 35;">9</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 1;">i</span><span class="nc684nl6 l94mrbxd l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 7;">c</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 13;">9</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 34;">9</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 16;">8</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 21;">9</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 32;">9</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 25;">0</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 15;">9</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4" style="order: 30;">h</span><span class="ihxqhq3m myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 38;">l</span><span class="l94mrbxd aenfhxwr myohyog2 b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 41;">5</span><span class="b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 14;">6</span><span class="t5a262vz l94mrbxd myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 29;">2</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 5;">9</span><span class="l94mrbxd aenfhxwr myohyog2 b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 28;">m</span><span class="b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 11;">6</span><span class="ihxqhq3m myohyog2 l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 20;">4</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4" style="order: 10;">1</span><span class="b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 17;">6</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 37;">1</span><span class="l94mrbxd aenfhxwr myohyog2 b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 3;">5</span><span class="l94mrbxd aenfhxwr myohyog2 b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 26;">5</span><span class="t5a262vz l94mrbxd myohyog2 l9j0dhe7 sdhka5h4" style="order: 22;">2</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 27;">0</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 8;">0</span><span class="l94mrbxd aenfhxwr myohyog2 b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 6;">5</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 36;">8</span><span class="t5a262vz aenfhxwr b6zbclly l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 4;">a</span><span class="t5a262vz nc684nl6 ihxqhq3m l94mrbxd aenfhxwr l9j0dhe7 sdhka5h4 orhKlDki sdHu" style="order: 19;">h</span></span></span></span></a>
```

Sorry about the code block. It is huge. It is important to point out that the aria-labelledby does match the correct id, I think the attribute is just too deep.

### Actual behavior:

A random string of letters and numbers is read.

### Expected behavior:

The value of the aria-labelledby item attached via id. In this example, the Facebook post timestamp.

#### NVDA installed/portable/running from source:
Installed.
#### NVDA version:
2022.1
#### Windows version:
Windows 10 Professional
#### Name and version of other software in use when reproducing the issue:
- Firefox: 101.0
- Google Chrome: 102.0.5005.63 (Official Build) (64-bit)
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No change in versions.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.

**Comment by seanbudd** [comment]
Welcome @alexstine ,

We are unable to process this issue as it stands.

To be able to reproduce this issue, we need a code sample. Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/).
This is very likely a problem with the implementation of the application/website, as a starting point please submit feedback to the authors to let them know about the problem.

Kind Regards,
NV Access Software Developers

**Comment by alexstine** [comment]
@alexarnaud Yes, I can still reproduce the problem. I am using Google Chrome.

The id I mentioned is referenced later in the code but I did not paste that snippet. The aria-labelledby is valid but I am still thinking it needs to be on the higher level link vs. the inner span. This is a sample of what I hear when browsing a group.

```
main landmark    Citizens of Irving News  visited  link    heading    level 1
Most Relevant  heading    level 2  button  
Lorena Yasmeen Romero  link    heading    level 2
link    eSosopntrd60tfft51t0913g7fag9ahllg814gc5l1   
graphic    Shared with Public group
menu button  collapsed  subMenu    Actions for this post
Hi everyone ! Does anyone know where they start hiring at the age of 15?
```

I cleared all Google Chrome/Firefox history/cache. It did not solve the problem.

This is likely a Facebook issue but I opened this issue to discuss if aria-labelledby should be working on inner spans vs. top level links.

Thanks.

**Comment by alexarnaud** [comment]
"aria-labelleby" is intended to provide or supersede a label. In this case.

Based on the information I've, it will only supersede the span label. The link label will be the result of all inner span. If one has an "aria-labelledby", it only means it will supersede just the content into the individual span with the "aria-labelledby" attribute.

Without a way to reproduce the issue, we wouldn't be able to investigate more. Maybe @mcking65 from Facebook could help?

**Comment by seanbudd** [comment]
Facebook appears to have fixed this bug

**Comment by alexstine** [comment]
@seanbudd Definitely not fixed for me.

> stprenoodSul4884091626c2t1856g605l219i86ha

I am still expecting something like:

> 4h

or similar. Not nonsense.

**Comment by alexarnaud** [comment]
Sorry @alexstine, the NVDA team has a lot of issue to handle, sometime it's hard to follow each one deeply.

Fortunately for us, Facebook provide a contact form to report accessibility issues here: https://www.facebook.com/help/contact/169372943117927

Don't hesitate to tell us if the issue is fixed or not. You can give the link of this issue to the Facebook team, I hope they could bring us data to let us determine if there is an issue on the NVDA side or not but it'd be more efficient to let the Facebook team investigate their code.

**Comment by alexstine** [comment]
@alexarnaud I've reported it to them several times and nothing. They fixed it and one week later, I was reporting it again. At least once a week I try to fill out that form. I'll give it another go. I just thought I'd open this to see if it was a bug but it does appear to be Facebook. JAWS has the same problem.

Unrelated question. I'm not much of a Python developer but I do work in DevOps. Is there any hope in the future to try to onboard new contributors? I really appreciate NVDA and I'd like to give back. The barrier of entry just seems so high. 😞 

Thanks.

**Comment by lukaszgo1** [comment]
> Unrelated question. I'm not much of a Python developer but I do work in DevOps. Is there any hope in the future to try to onboard new contributors? I really appreciate NVDA and I'd like to give back. The barrier of entry just seems so high. 

If there is something specific which stops you from attempting to contribute consider subscribing to [nvda-devel mailing list](https://groups.io/g/nvda-devel/) and someone may help you there.

**Comment by XLTechie** [comment]
> Unrelated question. I'm not much of a Python developer but I do work in DevOps. Is there any hope in the future to try to onboard new contributors? I really
> appreciate NVDA and I'd like to give back. The barrier of entry just seems so high. 😞

Why do you say that?

You can read the development guide.
Then look for an issue labeled good for new dev, and take a crack at it.

Or get your feet wet by writing an add-on.
There is even a separate add-on development guide, if you choose to go that way.

**Comment by alexstine** [comment]
@XLTechie The code base just looks so daunting. Maybe I just need to quit making up reasons and dive in. Good to know there are some good first dev issues.
